modules/storages/lib/open_project/storages/engine.rb

```diff
@@ -135,6 +135,21 @@ def self.external_file_permissions
         ) do |payload|
           ::Storages::HealthService.new(storage: payload[:storage]).healthy
         end
+
+        OpenProject::Notifications.subscribe(
+          ::OpenIDConnect::UserTokens::FetchService::TOKEN_OBTAINED
+        ) do |payload|
+          audience = payload[:audience]
+          token = payload[:token]
+          storage = Storages::Storage.all.find { |s| s.audience == audience }
```

Comment:
:green_circle: The number of storages will usually be low, so I don't care too much that we fetch all of them, just to be able to validate their audience. I guess the main problem is, that right now there is no audience "column" for one drive storages and thus we can't filter this on the database level?